pulser/simulation/simconfig.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2020 Pulser Development Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing_extensions import Literal, get_args
+
+import numpy as np
+import qutip
+
+
+NOISE_TYPES = Literal['doppler', 'amplitude', 'SPAM', 'dephasing']
+mass = 1.45e-25
+kb = 1.38e-23
+keff = 8.7
+
+
+@dataclass
```

Comment:
I think we should push for having the dataclass with `frozen=True`. It will make things a bit ugly, but only inside then hidden methods. When you make it frozen, you will no longer be able to change attributes, so, for example, you can't do `self.temperature *= 1.e-6`. However, you can cheat by doing `self.__dict__["temperature"] *= 1e-6` instead. It will look uglier, but it has the added benefit of not letting the user change the config without cheating themselves. See how we do this in `Variable`, for example.

Reply:
I found an uglier solution on stackexchange and thought it was ugly, but yours is way better. We'll use that one with `frozen=True`.